implementations/filter.js

```diff
@@ -0,0 +1,29 @@
+/**
+ *
+  filter() creates a new array with each element in the original
+  array if and only if the element pass the supplied predicate callback.
+
+  The predicate callback is invoked with three arguments:
+
+  - the value of the element
+  - the index of the element
+  - the Array object being traversed
+
+  The predicate callback is expected to return true if the element should
+  appear in the new array, otherwise false.
+
+  You can optionally specify a value to use as `this` in the callback
+  as the second argument to filter().
+
+  If no elements pass the predicate callback, an empty array will be returned
+*/
+
+Array.prototype.myFilter = function myFilter(callback, thisArg = this) {
```

Comment:
If a thisArg parameter is provided to filter, it will be used as the callback's this value.  Otherwise, the value undefined will be used as its this value.